### PR TITLE
[BUG FIX] Fixed bug on absolute path

### PR DESCRIPTION
**Pull request by MarcosFelipeBC**
Resolving bug pointed on issue [#6](https://github.com/JoaoDanielRufino/go-input-autocomplete/issues/6)

It was created also a main function to make testing easier and some methods on code to improve the program

### CODE REVIEW

File: autocomplete.go [hunk 1]
```diff
@@ -40,6 +40,13 @@ func (a autocomplete) linuxAutocomplete(path string) (string, error) {
 	for _, subPath := range lastValidSplittedPath {
 		lastValidPath += "/" + subPath
 	}
+	if lastValidPath == "" {
+		lastValidPath = "/"
+	}
+
+	if !a.cmd.IsDir(lastValidPath) {
```
Comment (joaorufinozup) [inline autocomplete.go:47]: I noticed that `a.cmd.IsDir()` and `isDir()` are functions that does the same logic. Could you just use one?

File: cmd.go [hunk 2]
```diff
@@ -26,3 +29,13 @@ func (c CmdLinux) ListContent(path string) ([]string, error) {
 
 	return lsOutput, nil
 }
+
+func (c CmdLinux) IsDir(dir string) bool {
```
Comment (joaorufinozup) [inline cmd.go:33]: `IsDir()` is a generic function, could you remove it from the struct CmdLinux and implement as a normal helper function?

File: cmd.go [hunk 3]
```diff
@@ -26,3 +29,13 @@ func (c CmdLinux) ListContent(path string) ([]string, error) {
...
+	info, err := os.Stat(dir)
+	if err != nil {
+		fmt.Println(err)
```
Comment (joaorufinozup) [inline cmd.go:36]: When you get an error, could you return the error instead of printing it?

**Review by joaorufinozup — CHANGES_REQUESTED** [review]

File: autocomplete.go (hunk 1 above)
Comment (JoaoDanielRufino) [inline autocomplete.go]: I noticed that `a.cmd.IsDir()` and `isDir()` are functions that does the same logic. Could you just use one?

File: cmd.go (hunk 2 above)
Comment (JoaoDanielRufino) [inline cmd.go]: `IsDir()` is a generic function, could you remove it from the struct `CmdLinux` and implement as a normal helper function?

File: cmd.go (hunk 3 above)
Comment (JoaoDanielRufino) [inline cmd.go]: When you get an error, could you return the error instead of printing it?

**Review by JoaoDanielRufino — CHANGES_REQUESTED** [review]

File: autocomplete.go (hunk 1 above)
Comment (MarcosFelipeBC) [reply autocomplete.go]: Sure! It was a mistake!

File: cmd.go (hunk 2 above)
Comment (MarcosFelipeBC) [reply cmd.go]: Sure!

File: cmd.go (hunk 3 above)
Comment (MarcosFelipeBC) [reply cmd.go]: Sure!

Comment (MarcosFelipeBC) [reply cmd.go]: I think that isDir should not be a blocking function, so what if we just ignore the error if it occur sometime?

Comment (JoaoDanielRufino) [reply cmd.go]: Ok, I agree

**Review by JoaoDanielRufino — APPROVED** [review]
Thanks for the contribution

**Comment by MarcosFelipeBC** [comment]
It was a pleasure! Can you set the label "hacktoberfest-accepted" to this pull request please?

**Pull request merged**